ZenPacks/zenoss/OpenStackInfrastructure/apiclients/txapiclient.py

```diff
@@ -77,39 +77,43 @@ def __init__(self, username, password, auth_url, project_id, is_admin=False):
 
     @property
     def keystone_endpoints(self):
-        if '/v2' in self.auth_url:
+        if '/v2' in self.auth_url and self.is_admin is not None:
             self._admin_only()
         self.user_agent = 'zenoss-keystoneclient'
         return self._apis.setdefault('keystone_endpoints', API(self, '/endpoints'))
 
     @property
     def keystone_tenants(self):
-        if '/v2' in self.auth_url:
-            self._admin_only()
         self.user_agent = 'zenoss-keystoneclient'
         return self._apis.setdefault('keystone_tenants', API(self, '/tenants'))
 
     @property
     def keystone_users(self):
-        if '/v2' in self.auth_url:
+        if '/v2' in self.auth_url and self.is_admin is not None:
             self._admin_only()
         self.user_agent = 'zenoss-keystoneclient'
         return self._apis.setdefault('keystone_users', API(self, '/users'))
 
     @property
     def keystone_roles(self):
-        if '/v2' in self.auth_url:
+        if '/v2' in self.auth_url and self.is_admin is not None:
             self._admin_only()
         self.user_agent = 'zenoss-keystoneclient'
         return self._apis.setdefault('keystone_roles', API(self, '/OS-KSADM/roles'))
 
     @property
     def keystone_services(self):
-        if '/v2' in self.auth_url:
+        if '/v2' in self.auth_url and self.is_admin is not None:
             self._admin_only()
         self.user_agent = 'zenoss-keystoneclient'
         return self._apis.setdefault('keystone_services', API(self, '/OS-KSADM/services'))
 
+    @property
+    def keystone_version(self):
+        # a very cheap keystone API call
+        self.user_agent = 'zenoss-keystoneclient'
```

Comment:
If you can see a safe way to do it, can we lower the timeout for this request, while leaving it at its default value for other ones?